### DISCUSSION TITLE
Can we add DNS rules support?

### DISCUSSION THREAD

**Discussion by Mishasama** [DNSCrypt/dnscrypt-proxy#2702] [Ideas]
I mean something like this.
![Rules](https://www.yogadns.com/screen2.png)
Different groups of servers correspond to different groups of websites to achieve a similar effect as the ECS Preferred Route.
e.g. For Ukrainian sites, I use Ukrainian DNS servers; for Russian sites, I use Russian DNS servers.

**Comment by lifenjoiner** [comment]
That's a completely different logic than the one we have now.

[Forwarding](https://github.com/DNSCrypt/dnscrypt-proxy/wiki/Forwarding) and using multiple instances is a possible way to gain the same effect.

**Reply by Mishasama** [reply]
> using multiple instances

This sounds extremely resource-intensive, and it appears that there is still no comparable product that can replace the proprietary software depicted in the image.
